packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -23,9 +22,21 @@ export 'binary_messenger.dart' show BinaryMessenger;
 export 'binding.dart' show RootIsolateToken;
 export 'message_codec.dart' show MessageCodec, MethodCall, MethodCodec;
 
-bool _debugProfilePlatformChannelsIsRunning = false;
-const Duration _debugProfilePlatformChannelsRate = Duration(seconds: 1);
-final Expando<BinaryMessenger> _debugBinaryMessengers = Expando<BinaryMessenger>();
+/// Profile and print statistics on Platform Channel usage.
+///
+/// When this is true statistics about the usage of Platform Channels will be
+/// printed out periodically to the console and Timeline events will show the
+/// time between sending and receiving a message (encoding and decoding time
+/// excluded).
+///
+/// The statistics include the total bytes transmitted and the average number of
+/// bytes per invocation in the last quantum. "Up" means in the direction of
+/// Flutter to the host platform, "down" is the host platform to flutter.
+const bool kProfilePlatformChannels = false;
```

Comment:
Would it help to make this a `const bool.fromEnvironment`?

I can't remember if that works well in google's build environment or not. It also may have issues on web.

Reply:
That could work and we could wire it up through the build rules to have something like:

```bzl
flutter_application(
  profilePlatformChannels = True,
)
```

But I'm not sure if this is so helpful in the near term. Since we probably don't want users to turn that flag on permanently, having to patch their `flutter_application` is very similar to having to patch this file locally when profiling. For now I'm working on documentation to do things manually, but in the long term this can ideally be some toggle in devtools.